src/StreamFactoryInterface.php

```diff
@@ -9,15 +9,11 @@
     /**
      * Create a new stream from a resource or a string.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
+     * String content will be used as the body of the stream. Files paths are not supported.
```

Comment:
> According to the dependency inversion principle, a library should not depend on a implementation (utility package), it should rely on interfaces (PSR-17)

For the functionality covered by those interfaces, of course. No one would disagree with that. What we disagree about is the scope.

> The interfaces in PSR-17 should be easy to use for a library author so no utility library is needed.

No other libraries *within what scope?*

Of course other libraries are going to be needed - you are still going to depend on a dispatcher, router, middleware, DI container, and whatever else (framework, components, etc.) is required for your project and choice of architecture, etc.

We're not going to cover all of things with abstractions, because we don't need to do that in order to make such components interoperable.

The point of having interfaces is interoperability, so we can abstract from the precise choice of implementations - so that components that comply with those interfaces can interoperate seamlessly.

> There could be competing PSR-17 implementations

There *will* be competing libraries as well: dispatchers, various middleware, and probably many other utilities and useful libraries made possible by the interoperability layer.

IMO, the scope of an interoperability layer must be kept low-level - if you introduce higher-order abstractions that aren't strictly necessary for interoperability, you're merely burdening *every* developer with the same implementations, tests, etc. while gaining *zero* in terms of functionality.

Interfaces are not a replacement for implementations - they're a means of making implementations interchangeable.

The only reason to go beyond the scope required for intereoperability, is to displace or suppress design diversity.

I hope we're not trying to do that?